### PR TITLE
Patch Settings Injection to work no matter the language

### PR DESCRIPTION
**Pull request by Julienraptor02**
fixes https://github.com/GooseMod/OpenAsar/issues/179

### CODE REVIEW

File: src/mainWindow.js [hunk 1]
```diff
@@ -41,12 +41,12 @@ setInterval(() => {
 
   host.insertAdjacentElement('afterend', oaVersion);
 
-  let advanced = document.querySelector('[class*="socialLinks"]').parentElement.querySelector('[class*="premiumTab"] ~ [class*="header"] + [class*="item"] + [class*="item"] + [class*="item"] + [class*="item"] + [class*="item"] + [class*="item"] + [class*="item"] + [class*="item"] + [class*="item"] ~ [class*="separator"]')?.previousSibling;
-  if (!advanced) advanced = [...document.querySelectorAll('[class*="item"]')].find(x => x.textContent === 'Advanced');
```
Comment (CanadaHonk) [inline src/mainWindow.js:45]: Keep this fallback?

File: src/mainWindow.js [hunk 2]
```diff
@@ -41,12 +41,12 @@ setInterval(() => {
...
+  let advanced = document.querySelector('[class^=side] :nth-last-child(1 of [class^=header]:not([class^=headerText])')?.previousElementSibling?.previousElementSibling;;
```
Comment (CanadaHonk) [inline src/mainWindow.js:44]: nit: `;;` -> `;`

File: src/mainWindow.js [hunk 3]
```diff
@@ -41,12 +41,12 @@ setInterval(() => {
...
   if (!advanced || document.getElementById('openasar-item')) return;
 
   const oaSetting = advanced.cloneNode(true);
   oaSetting.textContent = 'OpenAsar';
+  oaSetting.setAttribute('aria-label', 'OpenAsar');
```
Comment (CanadaHonk) [inline src/mainWindow.js:49]: Unnecessary as it is just a duplicate of text content

**Review by CanadaHonk — CHANGES_REQUESTED** [review]

File: src/mainWindow.js (hunk 3 above)
Comment (Julienraptor02) [reply src/mainWindow.js:49]: it's a duplicate, however if we don't do that, then an advanced tag is being left there
imo, either it should be changed to this or we could just remove the attribute as some have the attributes and some don't

File: src/mainWindow.js (hunk 2 above)
Comment (Julienraptor02) [reply src/mainWindow.js:44]: sry typo, will fix

File: src/mainWindow.js (hunk 1 above)
Comment (Julienraptor02) [reply src/mainWindow.js:45]: i think a fallback that's language exclusive doesn't make a lot of sense as there is already a fallback which is clicking on the version, but sure if you want, i could keep it

Comment (Julienraptor02) [reply src/mainWindow.js:45]: i think a fallback that's language exclusive doesn't make a lot of sense as there is already a fallback which is clicking on the version, but sure if you want, i could keep it

**Comment by Julienraptor02** [comment]
Note : while paying closer attention to the html, i noticed this new code had issues when client-mods like vencord inject their own settings section and they are in some special places

**Comment by Julienraptor02** [comment]
so the changes are bigges than i 1st expected
what i have done
1. implement the most simple method which is literally getting the Advanced tab directly
2. Implement back your fall back
3. Implement my new method as another fallback, because it has the advantage of resisting a change of amount of settings, but is broken by client-mods in some cases unlike your method
4. implement your en only way

imo 4 should go, because keeping smth that can work only for part of the users is a very bad idea and it's better to be broken for no one or everyone

2 and 3 could probably go too, but are decent fallbacks

in the rest of the changes
currently, i do setAttribute on the attributes that we clone that become wrong
those could be removeAttribute if you prefer
however, they should be there, otherwise, it mean that there is gonna be wrong misleading data


for the nit stuff, ik i'm recreating the vars, i have blocks instead of the statement being directly after the if and same for the return, but those can be fixed once you have decided what to do exactly

**Review by CanadaHonk — APPROVED** [review]
Looks good, thanks!

**Pull request merged**